### ISSUE TITLE
Enabling certain collectors in systems with multiple USB controllers causes no metrics to be returned

### ISSUE THREAD

**Issue by higels** [leoluk/perflib_exporter#9]
I'm running master which was cloned this morning compiled with go 1.11 running on a fully patched Server 2016 install.

The systems I've tested with have (a Dell PE FC630 and R730) have dual, identically named USB controllers like so:

<img width="354" alt="image" src="https://user-images.githubusercontent.com/1182646/45383441-15451f80-b5c0-11e8-8555-2c80c0b0ef1e.png">

If I enable certain perflib objects, I get no metrics returned, and an error message like the text below.

Specifically, object 6560 (Physical Network Interface Card Activity) causes this to happen, though it's possible other things do so as well. If I disable one of the "Standard Enhanced PCI to USB Host Controller" devices, it seems to run fine.

Exact CLI:

perflib_exporter.exe --telemetry.addr=":9100" --log.level="debug" --perflib.objects=2 --perflib.objects=4  --perflib.objects=234 --perflib.objects=238 --perflib.objects=510 --perflib.objects=546 --perflib.objects=6560

Though, this also happens with the default object list.

I've also noticed that even though I'm not requesting object 582 (ICMP), I still get a bunch of those, but don't seem to get any of those interface counters I was looking for. That's probably unrelated; let me know if you want to me file a separate issue.

```
45 error(s) occurred:
* collected metric "perflib_usb_bulk_bytes_total" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_isochronous_bytes_total" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_interrupt_bytes_total" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:3 > } was collected before with the same name and label values
* collected metric "perflib_usb_control_data_bytes_total" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:601 > } was collected before with the same name and label values
* collected metric "perflib_usb_controller_pci_interrupts_total" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:63 > } was collected before with the same name and label values
* collected metric "perflib_usb_controller_worksignals_total" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_bandwidth_used_for_interrupt" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_bandwidth_used_for_iso" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_iso_packet_errors_total" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_transfer_errors_total" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:4 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_idle" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:1 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_async_idle" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:2 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_async_cache_flush_count_total" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:3 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_periodic_idle" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:4 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_periodic_cache_flush_count_total" { label:<name:"name" value:"Standard Enhanced PCI to USB Host Controller" > counter:<value:5 > } was collected before with the same name and label values
* collected metric "perflib_usb_bulk_bytes_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_isochronous_bytes_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_interrupt_bytes_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_control_data_bytes_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:333 > } was collected before with the same name and label values
* collected metric "perflib_usb_controller_pci_interrupts_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_controller_worksignals_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_bandwidth_used_for_interrupt" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_bandwidth_used_for_iso" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_iso_packet_errors_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_transfer_errors_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:2 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_idle" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_async_idle" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_async_cache_flush_count_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_periodic_idle" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_periodic_cache_flush_count_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_bulk_bytes_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_isochronous_bytes_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_interrupt_bytes_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_control_data_bytes_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:361 > } was collected before with the same name and label values
* collected metric "perflib_usb_controller_pci_interrupts_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_controller_worksignals_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_bandwidth_used_for_interrupt" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_bandwidth_used_for_iso" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_iso_packet_errors_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_transfer_errors_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:2 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_idle" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_async_idle" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_async_cache_flush_count_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_periodic_idle" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
* collected metric "perflib_usb_host_controller_periodic_cache_flush_count_total" { label:<name:"name" value:"Generic USB Hub" > counter:<value:0 > } was collected before with the same name and label values
```

**Comment by leoluk** [comment]
Thank you for this bug report, I'm trying to reproduce this.

Could you send me the output of `/dump?query=6550`? 

> I've also noticed that even though I'm not requesting object 582 (ICMP), I still get a bunch of those, but don't seem to get any of those interface counters I was looking for. That's probably unrelated; let me know if you want to me file a separate issue.

Feel free to file a separate issue, definitely worth looking into. The perflib API seems to return all objects that a given provider DLL exposes when you request one of them.

**Comment by higels** [comment]
Here you go.

I'll file another issue once I've debugged it a bit more.

[perflib-exporter-6550.html.txt](https://github.com/leoluk/perflib_exporter/files/2373007/perflib-exporter-6550.html.txt)

**Comment by leoluk** [comment]
Please re-check with the latest master - part of this might be resolved by #14.

**Comment by leoluk** [comment]
Assuming this is fixed, please re-open if necessary
